test/modular/TraceurTestRunner.js

```diff
@@ -43,17 +43,10 @@ export class TraceurTestRunner extends Mocha6 {
   run() {
     let numberOfFailures = 0;
     return this.expandPatterns().then(() => {
-      return super.run().then((runner) => {
-        return new Promise((resolve, reject) => {
-          runner.on('fail', (err) => {
-            numberOfFailures++;
-          });
-          runner.on('end', () => {
-            resolve(numberOfFailures);
-          });
-          runner.on('error', (ex) => {
-            reject(ex);
-          });
+      // Convert from callback to promise
+      return new Promise((resolve, reject) => {
```

Comment:
I moved the conversion from callbacks to promises up from this class to `Mocha6.run()` where it belongs.